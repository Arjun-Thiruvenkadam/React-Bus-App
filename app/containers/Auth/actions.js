/*
 *
 * Auth actions
 *
 */

import * as actionTypes from './constants';

export function inputChange(event, controlName) {
  return {
    type: actionTypes.INPUT_CHANGE,
    event,
    controlName,
  };
}

export function submit(event) {
  return {
    type: actionTypes.SUBMIT,
    event,
  };
}

export function startLoading() {
  return {
    type: actionTypes.START_LOADING,
  };
}

export function stopLoading() {
  return {
    type: actionTypes.STOP_LOADING,
  };
}

export function authSuccess(name, token) {
  return {
    type: actionTypes.AUTH_SUCCESS,
    name,
    token,
  };
}

export function switchAuthMode(){
  return{
    type: actionTypes.SWITCH_AUTHMODE,
  }
}

export function logout(){
  return{
    type: actionTypes.LOGOUT,
  }
}

export function invalid(){
  return{
    type: actionTypes.INVALID,
  }
}
