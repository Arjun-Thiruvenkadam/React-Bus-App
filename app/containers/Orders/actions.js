/*
 *
 * Orders actions
 *
 */

import * as actionTypes from './constants';

export function bookTicket() {
  return {
    type: actionTypes.BOOK_TICKET,
  };
}

export function showSpinner() {
  return {
    type: actionTypes.SHOW_SPINNER,
  };
}

export function hideSpinner() {
  return {
    type: actionTypes.HIDE_SPINNER,
  };
}

export function saveTickets(tickets) {
  return {
    type: actionTypes.SAVE_TICKETS,
    tickets,
  };
}

export function cancelOrder() {
  return {
    type: actionTypes.CANCEL_ORDER,
  };
}

export function selectTicket(ticketId) {
  return {
    type: actionTypes.SELECT_TICKET,
    ticketId,
  };
}

export function fetchData() {
  return {
    type: actionTypes.FETCH_DATA,
  };
}

export function completeBooking() {
  return {
    type: actionTypes.COMPLETE_BOOKING,
  };
}

export function clearSelection() {
  return {
    type: actionTypes.CLEAR_SELECTION,
  };
}

export function update() {
  return {
    type: actionTypes.UPDATE,
  };
}
