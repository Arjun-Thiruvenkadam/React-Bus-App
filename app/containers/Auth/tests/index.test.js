import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { browserHistory } from 'react-router-dom';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import ConnectedAuth, { mapDispatchToProps, Auth } from '../index';
import * as actions from '../actions';
import configureStore from '../../../configureStore';

describe('<Auth />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ConnectedAuth />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot with spinner', () => {
    const auth = {
      controls: {
        name: {
          elementType: 'input',
          elementConfig: {
            type: 'text',
            placeholder: 'Name',
          },
          value: 'data',
          validation: {
            required: true,
            minLength: 3,
          },
          valid: true,
          touched: true,
        },
        email: {
          elementType: 'input',
          elementConfig: {
            type: 'email',
            placeholder: 'Mail Address',
          },
          value: 'test@gmail.com',
          validation: {
            required: true,
            isEmail: true,
          },
          valid: true,
          touched: true,
        },
        password: {
          elementType: 'input',
          elementConfig: {
            type: 'password',
            placeholder: 'Password',
          },
          value: 'password',
          validation: {
            required: true,
            minLength: 6,
          },
          valid: true,
          touched: true,
        },
      },
      isSignup: true,
      loginToken: false,
      userId: '',
      loading: true,
      authenticated: false,
    };
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <Auth auth={auth} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot with login', () => {
    const auth = {
      controls: {
        name: {
          elementType: 'input',
          elementConfig: {
            type: 'text',
            placeholder: 'Name',
          },
          value: 'data',
          validation: {
            required: true,
            minLength: 3,
          },
          valid: true,
          touched: true,
        },
        email: {
          elementType: 'input',
          elementConfig: {
            type: 'email',
            placeholder: 'Mail Address',
          },
          value: 'test@gmail.com',
          validation: {
            required: true,
            isEmail: true,
          },
          valid: true,
          touched: true,
        },
        password: {
          elementType: 'input',
          elementConfig: {
            type: 'password',
            placeholder: 'Password',
          },
          value: 'password',
          validation: {
            required: true,
            minLength: 6,
          },
          valid: true,
          touched: true,
        },
      },
      isSignup: false,
      loginToken: false,
      userId: '',
      loading: true,
      authenticated: true,
    };
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <MemoryRouter>
          <IntlProvider locale="en">
            <Auth auth={auth} />
          </IntlProvider>
        </MemoryRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('inputChangeHandler', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.inputChangeHandler).toBeDefined();
      });

      it('should dispatch inputChange when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const data = 'testing';
        const controlName = 'name';
        result.inputChangeHandler(data, controlName);
        expect(dispatch).toHaveBeenCalledWith(
          actions.inputChange(data, controlName),
        );
      });
    });

    describe('submitHandler', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.submitHandler).toBeDefined();
      });

      it('should dispatch submit when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const data = 'testing';
        result.submitHandler(data);
        expect(dispatch).toHaveBeenCalledWith(actions.submit(data));
      });
    });

    describe('switchAuthModeHandler', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.switchAuthModeHandler).toBeDefined();
      });

      it('should dispatch switchAuthMode when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.switchAuthModeHandler();
        expect(dispatch).toHaveBeenCalledWith(actions.switchAuthMode());
      });
    });
  });
});
