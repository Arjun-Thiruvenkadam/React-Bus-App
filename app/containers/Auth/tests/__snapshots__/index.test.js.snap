// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> should render and match the snapshot 1`] = `
<div
  class="Auth"
>
  <h2>
    Login to start your journey
  </h2>
  <form>
    <div
      class="Input"
    >
      <label
        class="Label"
      />
      <input
        class="InputElement"
        placeholder="Mail Address"
        type="email"
        value=""
      />
    </div>
    <div
      class="Input"
    >
      <label
        class="Label"
      />
      <input
        class="InputElement"
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <button
      class="Button Disabled"
      disabled=""
    >
      SUBMIT
    </button>
  </form>
  <button
    class="Button Danger"
  >
    SWITCH TO 
    SIGNUP
  </button>
  <br />
  <br />
</div>
`;

exports[`<Auth /> should render and match the snapshot with login 1`] = `
<div
  class="Auth"
>
  <h2>
    Login to start your journey
  </h2>
  <form>
    <div
      class="Loader"
    >
      <h3>
        <span>
          Loading...
        </span>
      </h3>
    </div>
    <button
      class="Button Success"
    >
      SUBMIT
    </button>
  </form>
  <button
    class="Button Danger"
  >
    SWITCH TO 
    SIGNUP
  </button>
  <br />
  <h3
    style="color: red;"
  >
    Check email and password
  </h3>
  <br />
</div>
`;

exports[`<Auth /> should render and match the snapshot with spinner 1`] = `
<div
  class="Auth"
>
  <h2>
    Don't have an account? Sign up
  </h2>
  <form>
    <div
      class="Loader"
    >
      <h3>
        <span>
          Loading...
        </span>
      </h3>
    </div>
    <button
      class="Button Success"
    >
      SUBMIT
    </button>
  </form>
  <button
    class="Button Danger"
  >
    SWITCH TO 
    LOGIN
  </button>
  <br />
  <h3
    style="color: red;"
  >
    Email already exists
  </h3>
  <br />
</div>
`;
